src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -190,6 +197,19 @@ class IntervalExpressionSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class ArrayLiteralSegment(BaseSegment):
```

Comment:
This addresses a new issue related to UDFs, #337. UDFs can take `ARRAY` parameters, but this is needed in order to allow passing an array literal when calling a function. This syntax is BigQuery specific, but similar to a syntax supported by Postgres. See the linked issue for details.